### PR TITLE
fix(user-paths): fix user paths cleaning filters (fixes #7326, #7327)

### LINKED ISSUE #7326: Empty path cleaning rule renders weird
## Bug description

We have a path cleaning rule to strip the domain from out paths. This is the rule we are using:

image

In the settings, this is how it renders:

image

## Expected behavior

It renders as it's own "chip" / "pill".

## How to reproduce

1. Create a path cleaning rule with an empty alias

## Environment

- [ ] PostHog Cloud
- [x] self-hosted PostHog (ClickHouse-based), version/commit: 1.30.0
- [ ] self-hosted PostHog (Postgres-based, legacy), version/commit: _please provide_

### PR DESCRIPTION
**Pull request by thmsobrmlr**
## Problem/Changes

This PR 
- fixes an issue where empty path cleaning rules would render as button instead of "pill" https://github.com/PostHog/posthog/issues/7326
- fixes an issue where editing an existing path cleaning rule would instead create a new one https://github.com/PostHog/posthog/issues/7327
- adds a lemon pill component (see also here https://github.com/PostHog/posthog/issues/9235#issuecomment-1400183861)
- adds some missing storybook documentation

Follow up (not now):
- https://github.com/PostHog/posthog/issues/13872
- https://github.com/PostHog/posthog/issues/13876

## How did you test this code?

Manually verified the add/edit/delete flow for local and global path grouping filters.

### CODE REVIEW

File: frontend/src/lib/components/LemonPill/LemonPill.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { LemonButton } from '../LemonButton/LemonButton'
+import { IconClose } from '../icons'
+
+export type LemonPillProps = {
+    children: React.ReactNode
+    onClick: () => void
+    onDelete: () => void
+}
+
+export function LemonPill({ children, onClick, onDelete }: LemonPillProps): JSX.Element {
```
Comment (Twixes) [inline frontend/src/lib/components/LemonPill/LemonPill.tsx:10]: What's the difference between this and a rounded `LemonSnack`? They _seem_ quite similar

**Review by Twixes — COMMENTED** [review]
Looks good but I'm a bit wary of adding `LemonPill`, since the `LemonTag` and `LemonSnack` story already is a bit muddy 😅

File: frontend/src/styles/utilities.scss [hunk 2]
```diff
@@ -731,6 +731,9 @@
 .rounded-3xl {
     border-radius: calc(var(--radius) * 6); /* 24px */
 }
+.rounded-4xl {
```
[the remark this answers is not in the document]
Comment (thmsobrmlr) [reply frontend/src/styles/utilities.scss:734]: Unfortunately `rounded-full` rounds more than `rounded-4xl`. Tailwind doesn't have `rounded-4xl`, but can be customized to include it. I can use inline styles as alternative as well.

File: frontend/src/lib/components/LemonPill/LemonPill.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    return (
+        <div
+            className="h-8 px-4 py-1 rounded-4xl flex items-center bg-primary-alt-highlight text-primary-alt cursor-pointer"
+            onClick={onClick}
+        >
+            {children}
+            <LemonButton
+                className="ml-1"
```
[the remark this answers is not in the document]
Comment (thmsobrmlr) [reply frontend/src/lib/components/LemonPill/LemonPill.tsx:18]: Makes sense.

File: frontend/src/lib/components/LemonPill/LemonPill.tsx (hunk 1 above)
Comment (thmsobrmlr) [reply frontend/src/lib/components/LemonPill/LemonPill.tsx:10]: Yep, exactly. There's not much difference between them. I'll refactor LemonSnack and add a `type`-prop with options `regular` and `pill`.

**Comment by thmsobrmlr** [comment]
I've added a `pill` type for the `LemonSnack` component:

Screenshot 2023-01-25 at 01 02 09
Screenshot 2023-01-25 at 01 05 05

**Review by Twixes — APPROVED** [review]
👌

**Pull request merged**